scarb/tests/package.rs

```diff
@@ -102,11 +104,13 @@ impl PackageChecker {
         self
     }
 
-    fn read_file(&self, path: impl AsRef<Path>) -> &str {
+    fn read_file(&self, path: impl AsRef<Path>) -> String {
```

Comment:
Why this change? 🤔

Reply:
I've changed `actual_files` from `HashMap<PathBuf, String>,` to `HashMap<PathBuf, Vec<u8>>` so it can store files that are not valid utf-8 Strings - i.e. the dll binaries. It has to be stored in memory, as it's unpacked from a tarball without extracting to disk. 

Then I just converted it back to utf-8 string with `to_string_loosly`, which was the first thing that came to my mind, which requires allocations (thus String instead of &str). 
Yes, it can still be `str` when used with `std::str::from_utf8` - changed it back. It does not really matter here (it's just one allocation, just tests), but you are right it can be avoided :D .